php-frontend/src/main/java/org/sonar/plugins/php/api/symbols/QualifiedName.java

```diff
@@ -28,11 +29,15 @@
 public interface QualifiedName {
 
   /**
-   * @return the last element of qualified name i.e. {@code bar} for {@code namespace\foo\bar} , or {@code method} for {@code namespace\A::method}
-   *
+   * @return the last element of qualified name i.e. {@code bar} for {@code namespace\foo\bar}, or {@code method} for {@code namespace\A::method}
    */
   String simpleName();
 
+  /**
+   * @return the parent qualified name if it exists i.e {@code foo} for {@code foo\bar}, or {@code foo} for {@code foo\A::method}
```

Comment:
`{@code foo} for {@code foo\A::method}` is not what's implemented.
Three alternative ideas:
* Move `parent()` to SymbolQualifiedName (non-public API): that requires a cast in `UncatchableExceptionCheck`.
* Rename `parent()` to `owner()` and make MemberQualifiedName return `foo\A` for `foo\A::method`.
* Undo those changes to `QualifiedName` and go back to a string operation.